src/V3String.cpp

```diff
@@ -380,12 +380,56 @@ void VHashSha256::selfTest() {
 // VName
 
 string VName::dehash(const string& in) {
-    const string::size_type pos = in.find("__Vhsh");
-    if (VL_LIKELY(pos == string::npos)) return in;
-    const string vhsh = in.substr(pos);
-    const auto& it = s_dehashMap.find(vhsh);
-    UASSERT(it != s_dehashMap.end(), "String not in reverse hash map '" << vhsh << "'");
-    return in.substr(0, pos) + it->second;
+    static const char VHSH[] = "__Vhsh";
+
+    std::string dehashed;
+
+    // Need to split 'in' into components separated by __DOT__, 'last_dot_pos'
+    // keeps track of the position after the most recently found instance of __DOT__
+    string::size_type last_dot_pos = 0;
+    do {
+        // Find the position of the next dot.
+        const string::size_type next_dot_pos = in.find("__DOT__", last_dot_pos);
+
```

Comment:
```suggestion
        const string::size_type next_dot_pos = in.find("__DOT__", last_dot_pos);
```

Good commenting, but this one isn't needed as is just implied by the code.